devimint/src/main.rs

```diff
@@ -889,6 +889,79 @@ async fn lightning_gw_reconnect_test(dev_fed: DevFed, process_mgr: &ProcessManag
     Ok(())
 }
 
+async fn gw_reboot_test(dev_fed: DevFed, process_mgr: &ProcessManager) -> Result<()> {
+    #[allow(unused_variables)]
+    let DevFed {
+        bitcoind,
+        cln,
+        lnd,
+        fed,
+        gw_cln,
+        gw_lnd,
+        electrs,
+        esplora,
+        faucet,
+    } = dev_fed;
+
+    // Query current gateway infos
+    let mut cln_cmd = cmd!(gw_cln, "info");
+    let mut lnd_cmd = cmd!(gw_lnd, "info");
+    let (cln_value, lnd_value) = tokio::try_join!(cln_cmd.out_json(), lnd_cmd.out_json())?;
+
+    // Drop references to cln and lnd gateways so the test can kill them
+    drop(gw_cln);
+    drop(gw_lnd);
```

Comment:
Does this explicitly kill the lnd gateway process? For the lightning reconnect test, we actually need to call `stop_lightning_node` to kill the process

Reply:
this only kills gatewayd but preserves the lightning node processes. it then respawns gatewayd and provides handles to lightning

Reply:
So `drop` kills the gateway process then?

Reply:
true. thats the intent here